homeassistant/components/freedompro/__init__.py

```diff
@@ -0,0 +1,84 @@
+"""Support for freedompro."""
+from datetime import timedelta
+import logging
+
+from pyfreedompro import get_list, get_states
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_API_KEY
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import aiohttp_client
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS = ["light"]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up Freedompro from a config entry."""
+    hass.data.setdefault(DOMAIN, {})
+    api_key = entry.data[CONF_API_KEY]
+
+    coordinator = FreedomproDataUpdateCoordinator(hass, api_key)
+    await coordinator.async_config_entry_first_refresh()
+
+    entry.async_on_unload(entry.add_update_listener(update_listener))
+
+    hass.data[DOMAIN][entry.entry_id] = coordinator
+
+    hass.config_entries.async_setup_platforms(entry, PLATFORMS)
+
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Unload a config entry."""
+    unload_ok = await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
+    if unload_ok:
+        hass.data[DOMAIN].pop(entry.entry_id)
+
+    return unload_ok
+
+
+async def update_listener(hass, config_entry):
+    """Update listener."""
+    await hass.config_entries.async_reload(config_entry.entry_id)
+
+
+class FreedomproDataUpdateCoordinator(DataUpdateCoordinator):
+    """Class to manage fetching Freedompro data API."""
+
+    def __init__(self, hass, api_key):
+        """Initialize."""
+        self._hass = hass
+        self._api_key = api_key
+        self._devices = None
+
+        update_interval = timedelta(minutes=1)
+        super().__init__(hass, _LOGGER, name=DOMAIN, update_interval=update_interval)
+
+    async def _async_update_data(self):
```

Comment:
You might want to add an additional method to this class:

```py
    async def async_added_to_hass(self) -> None:
        """When entity is added to hass."""
        await super().async_added_to_hass()
        self._handle_coordinator_update()
```

This will update the state of the entity when it is added the Home Assistant.